### DISCUSSION TITLE
Exemplary Document Engineering

### DISCUSSION THREAD

**Discussion by orcmid** [amyjko/foundations-of-information#8] [General]
This effort strikes me as a great example of document engineering supported by GitHub, Markdown, etc.

I have a couple of questions about the document production process that I couldn't tease out by exploring the repository.

 1. What is the process that deploys (i.e., publishes) the book presentation at https://faculty.washington.edu/ajko/books/foundations-of-information/#/ ?
 2. As part of that, what or where is the association of image references in the chapter/ pages with the images located in the repository images/ folder.

I'm interested in this because I am intrigued by document engineering that is as classy as this as something to introduce to others who want to use GitHub as an instrument of document engineering by writers, not programmers.

**Comment by amyjko** [comment]
Great questions! I couldn't find a platform for creating online books that 1) was minimal infrastructure, 2) feature rich (e.g., search, index, flexible markup), 3) not ugly, so I made one:

https://github.com/amyjko/peruse

The basic infrastructure for a book includes:

1. Write chapters in Peruse-flavored markdown
2. Write a `book.json` spec that organizes the book, it's metadata, and its citations
3. Everything else is front end, no compiling, building, or other infrastructure. So the bundle is the chapters, the `book.json`, any images, and some boilerplate customizable HTML and CSS. Authors only have to focus on content.
4. Deployment is just a matter of hosting the HTML, CSS, content somewhere with a standard web server. I just host it on my UW faculty Apache server, but it could be on any web server with no additional infrastructure.

What's really missing from Peruse for writers is some web service with interfaces for editing book metadata and writing chapters. I haven't had the cycles to do build that, but it would definitely make it easier to write books!
